presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java

```diff
@@ -24,12 +25,14 @@ public class AccessControlContext
     private final QueryId queryId;
     private final Optional<String> clientInfo;
     private final Optional<String> source;
+    private final WarningCollector warningCollector;
 
-    public AccessControlContext(QueryId queryId, Optional<String> clientInfo, Optional<String> source)
+    public AccessControlContext(QueryId queryId, Optional<String> clientInfo, Optional<String> source, WarningCollector warningCollector)
```

Comment:
curious why AccessControlContext needs the warningCollector as well.

Reply:
SqlStandardAccessControl methods create MetastoreContext.. and AccessControlContext is the way to pipe the warningCollector till there.